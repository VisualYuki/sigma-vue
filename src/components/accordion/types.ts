import type {Ref} from 'vue'

export interface AccordionItemProps {
	_disabled?: string
	disabled?: boolean
	name: string
}

export interface AccordionProps {
	multiply?: boolean
}

export type MainContext = {
	modelValue: Ref<string[]>
	toggle(name: string): void
}
