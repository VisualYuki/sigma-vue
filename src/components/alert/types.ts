export type AlertProps = {
	closable?: boolean
	color?: 'error' | 'info' | 'primary' | 'success' | 'warning'
	description?: string
	title?: string
	variant?: 'solid'
}
