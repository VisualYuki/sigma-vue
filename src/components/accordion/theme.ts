import {cva} from 'class-variance-authority'

export const accordionStyles = {
	root: cva(['w-full'])
}

export const accordionItemStyles = {
	root: cva(['group']),
	trigger: cva(['flex items-center grow-1 gap-1.5']),
	content: cva([]),
	label: cva(['text-start break-words']),
	trailingIcon: cva(['ms-auto shrink-0 size-4 transition-transform duration-200 ' + String.raw`group-[.ui-accordion-item\_active]:rotate-90`])
}

// 	disabled: {
// 			true: {
// 				trigger: 'cursor-not-allowed opacity-75'
// 			}
// 		}

// export const accordionItemStyles = cva({
// 	slots: {
// 		item: 'group border-b border-(--ui-border) last:border-b-0',
// 		trigger: 'flex grow-1 flex items-center gap-1.5 font-medium text-sm py-3.5 cursor-pointer',
// 		content: '',
// 		body: 'text-sm pb-3.5',
// 		leadingIcon: 'shrink-0 size-5',
// 		trailingIcon: 'ms-auto shrink-0 size-4 transition-transform duration-200 ' + String.raw`group-[.ui-accordion-item\_active]:rotate-90`,
// 		label: 'text-start break-words'
// 	},
// 	variants: {
// 		disabled: {
// 			true: {
// 				trigger: 'cursor-not-allowed opacity-75'
// 			}
// 		}
// 	}
// })
