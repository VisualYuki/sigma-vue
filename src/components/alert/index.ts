import {defineAsyncComponent} from 'vue'
import Alert from './alert.vue'

export const UiAlert = Alert
export const UiAsyncAlert = defineAsyncComponent(() => import('./alert.vue'))
