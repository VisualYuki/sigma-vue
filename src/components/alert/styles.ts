/* eslint-disable perfectionist/sort-objects */

import {cva} from 'class-variance-authority'

export const alertStyles = {
	root: cva('relative overflow-hidden w-full rounded-md p-4 disabled', {
		variants: {
			color: {
				error: '',
				info: '',
				primary: '',
				success: '',
				warning: ''
			},
			variant: {
				solid: '',
				transparent: ''
			}
		},
		compoundVariants: [
			{
				class: 'bg-red-500 text-white',
				color: 'error',
				variant: 'solid'
			},
			{
				class: 'border border-red-500 text-red-500',
				color: 'error',
				variant: 'transparent'
			},
			{
				class: 'bg-blue-500 text-white',
				color: 'primary',
				variant: 'solid'
			},
			{
				class: 'border border-blue-500 text-blue-500',
				color: 'primary',
				variant: 'transparent'
			},
			{
				class: 'bg-green-500 text-white',
				color: 'success',
				variant: 'solid'
			},
			{
				class: 'border border-green-500 text-green-500',
				color: 'success',
				variant: 'transparent'
			},
			{
				class: 'bg-yellow-500 text-white',
				color: 'warning',
				variant: 'solid'
			},
			{
				class: 'border border-yellow-500 text-yellow-500',
				color: 'warning',
				variant: 'transparent'
			},
			{
				class: 'bg-gray-500 text-white',
				color: 'info',
				variant: 'solid'
			},
			{
				class: 'border border-gray-500 text-gray-500',
				color: 'info',
				variant: 'transparent'
			}
		]
	}),
	wrapper: cva('flex items-center'),
	close: cva('absolute top-3 right-3 cursor-pointer'),
	content: cva('flex flex-col gap-1'),
	title: cva('text-base font-medium pe-6'),
	description: cva('text-sm opacity-90')
}
