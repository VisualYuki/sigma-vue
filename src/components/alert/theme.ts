import type {ColorType} from '@/types/configuration'
export const alertVariants = ['solid', 'outline', 'soft', 'subtle'] as const
type VariantType = (typeof alertVariants)[number]

export default {
	slots: {
		root: 'relative overflow-hidden w-full rounded-(--ui-radius) p-4',
		wrapper: 'flex items-center',
		content: 'flex flex-col gap-1',
		withIcon: 'me-4',
		title: 'text-sm font-medium pe-6',
		description: 'text-sm opacity-90',
		close: 'absolute top-3 right-3 cursor-pointer'
	},
	variants: {
		color: {
			primary: '',
			secondary: '',
			success: '',
			info: '',
			warning: '',
			error: ''
		},
		variant: {
			solid: '',
			outline: '',
			soft: '',
			subtle: ''
		},
		multiline: {
			true: {
				root: 'items-start',
				actions: 'items-start mt-1'
			},
			false: {
				root: 'items-center',
				actions: 'items-center'
			}
		},
		center: {
			true: {
				wrapper: 'justify-center'
			}
		}
	},
	compoundVariants: [
		{
			color: 'primary' as ColorType,
			variant: 'solid' as VariantType,
			class: {
				root: 'bg-[var(--ui-primary)] text-[var(--ui-bg)]'
			}
		},
		{
			color: 'secondary' as ColorType,
			variant: 'solid' as VariantType,
			class: {
				root: 'bg-[var(--ui-secondary)] text-[var(--ui-bg)]'
			}
		},
		{
			color: 'success' as ColorType,
			variant: 'solid' as VariantType,
			class: {
				root: 'bg-[var(--ui-success)] text-[var(--ui-bg)]'
			}
		},
		{
			color: 'info' as ColorType,
			variant: 'solid' as VariantType,
			class: {
				root: 'bg-[var(--ui-info)] text-[var(--ui-bg)]'
			}
		},
		{
			color: 'warning' as ColorType,
			variant: 'solid' as VariantType,
			class: {
				root: 'bg-[var(--ui-warning)] text-[var(--ui-bg)]'
			}
		},
		{
			color: 'error' as ColorType,
			variant: 'solid' as VariantType,
			class: {
				root: 'bg-[var(--ui-error)] text-[var(--ui-bg)]'
			}
		},
		{
			color: 'primary' as ColorType,
			variant: 'outline' as VariantType,
			class: {
				root: 'text-[var(--ui-primary)] ring ring-inset ring-[var(--ui-primary)]/25'
			}
		},
		{
			color: 'secondary' as ColorType,
			variant: 'outline' as VariantType,
			class: {
				root: 'text-[var(--ui-secondary)] ring ring-inset ring-[var(--ui-secondary)]/25'
			}
		},
		{
			color: 'success' as ColorType,
			variant: 'outline' as VariantType,
			class: {
				root: 'text-[var(--ui-success)] ring ring-inset ring-[var(--ui-success)]/25'
			}
		},
		{
			color: 'info' as ColorType,
			variant: 'outline' as VariantType,
			class: {
				root: 'text-[var(--ui-info)] ring ring-inset ring-[var(--ui-info)]/25'
			}
		},
		{
			color: 'warning' as ColorType,
			variant: 'outline' as VariantType,
			class: {
				root: 'text-[var(--ui-warning)] ring ring-inset ring-[var(--ui-warning)]/25'
			}
		},
		{
			color: 'error' as ColorType,
			variant: 'outline' as VariantType,
			class: {
				root: 'text-[var(--ui-error)] ring ring-inset ring-[var(--ui-error)]/25'
			}
		},
		{
			color: 'primary' as ColorType,
			variant: 'soft' as VariantType,
			class: {
				root: 'bg-[var(--ui-primary)]/10 text-[var(--ui-primary)]'
			}
		},
		{
			color: 'secondary' as ColorType,
			variant: 'soft' as VariantType,
			class: {
				root: 'bg-[var(--ui-secondary)]/10 text-[var(--ui-secondary)]'
			}
		},
		{
			color: 'success' as ColorType,
			variant: 'soft' as VariantType,
			class: {
				root: 'bg-[var(--ui-success)]/10 text-[var(--ui-success)]'
			}
		},
		{
			color: 'info' as ColorType,
			variant: 'soft' as VariantType,
			class: {
				root: 'bg-[var(--ui-info)]/10 text-[var(--ui-info)]'
			}
		},
		{
			color: 'warning' as ColorType,
			variant: 'soft' as VariantType,
			class: {
				root: 'bg-[var(--ui-warning)]/10 text-[var(--ui-warning)]'
			}
		},
		{
			color: 'error' as ColorType,
			variant: 'soft' as VariantType,
			class: {
				root: 'bg-[var(--ui-error)]/10 text-[var(--ui-error)]'
			}
		},
		{
			color: 'primary' as ColorType,
			variant: 'subtle' as VariantType,
			class: {
				root: 'bg-[var(--ui-primary)]/10 text-[var(--ui-primary)] ring ring-inset ring-[var(--ui-primary)]/25'
			}
		},
		{
			color: 'secondary' as ColorType,
			variant: 'subtle' as VariantType,
			class: {
				root: 'bg-[var(--ui-secondary)]/10 text-[var(--ui-secondary)] ring ring-inset ring-[var(--ui-secondary)]/25'
			}
		},
		{
			color: 'success' as ColorType,
			variant: 'subtle' as VariantType,
			class: {
				root: 'bg-[var(--ui-success)]/10 text-[var(--ui-success)] ring ring-inset ring-[var(--ui-success)]/25'
			}
		},
		{
			color: 'info' as ColorType,
			variant: 'subtle' as VariantType,
			class: {
				root: 'bg-[var(--ui-info)]/10 text-[var(--ui-info)] ring ring-inset ring-[var(--ui-info)]/25'
			}
		},
		{
			color: 'warning' as ColorType,
			variant: 'subtle' as VariantType,
			class: {
				root: 'bg-[var(--ui-warning)]/10 text-[var(--ui-warning)] ring ring-inset ring-[var(--ui-warning)]/25'
			}
		},
		{
			color: 'error' as ColorType,
			variant: 'subtle' as VariantType,
			class: {
				root: 'bg-[var(--ui-error)]/10 text-[var(--ui-error)] ring ring-inset ring-[var(--ui-error)]/25'
			}
		}
	],
	defaultVariants: {
		color: 'primary' as ColorType,
		variant: 'solid' as VariantType
	}
}
