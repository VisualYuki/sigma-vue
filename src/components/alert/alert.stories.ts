import {type Meta, type StoryObj} from '@storybook/vue3'
import {UiAlert} from './'
import {colors} from '@/types/configuration'
import {alertVariants} from './theme'

const meta = {
	title: 'Components/Alert',
	component: UiAlert,
	tags: ['autodocs']
} satisfies Meta<typeof UiAlert>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
	argTypes: {
		title: {
			control: 'text'
		},
		description: {
			control: 'text'
		},
		color: {
			control: 'select',
			options: colors
		},
		variant: {
			control: 'select',
			options: alertVariants
		},
		closable: {
			control: 'boolean'
		},
		withIcon: {
			control: 'boolean'
		},
		center: {
			control: 'boolean'
		},
		modelValue: {
			control: 'boolean'
		}
	},
	args: {
		title: 'alert title',
		description: 'alert description',
		color: 'primary',
		variant: 'solid',
		closable: true,
		withIcon: false,
		center: false,
		modelValue: true
	},
	render: (args) => ({
		components: {
			UiAlert
		},
		setup() {
			return {args}
		},
		template: `
			<UiAlert v-bind='args'>
				<template #title> {{args.title}} </template>
				<template #description> {{args.description}} </template>
			</UiAlert>
		`
	})
}
