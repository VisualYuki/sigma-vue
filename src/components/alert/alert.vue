<template>
	<CollapseRoot v-model="modelValue">
		<CollapseContent>
			<div :class="alertStyles.root({color: props.color, variant: props.variant})" :data-name="ns.b()">
				<div :class="alertStyles.wrapper()" :data-name="ns.e('wrapper')">
					<div :class="alertStyles.content()" :data-name="ns.e('content')">
						<div v-if="props.title || $slots.title" :class="alertStyles.title()">
							<slot name="title">
								{{ props.title }}
							</slot>
						</div>
						<div v-if="props.description || $slots.description" :class="alertStyles.description()">
							<slot name="description">
								{{ props.description }}
							</slot>
						</div>
					</div>

					<CollapseTrigger v-if="props.closable" :class="alertStyles.close()">
						<Button @click="methods.handleClose">
							<Icon icon="mdi:close" size="20" />
						</Button>
					</CollapseTrigger>
				</div>
			</div>
		</CollapseContent>
	</CollapseRoot>
</template>

<script lang="ts" setup>
	import {useNamespace} from '@/utils'

	import {Button} from '../button'
	import {CollapseContent, CollapseRoot, CollapseTrigger} from '../collapse'
	import {Icon} from '../icon'
	import {alertStyles} from './styles'
	import {type AlertProps} from './types'

	const props = withDefaults(defineProps<AlertProps>(), {
		closable: false,
		color: 'primary',
		description: '',
		title: '',
		variant: 'solid'
	})

	const emit = defineEmits(['close'])

	const ns = useNamespace('alert')
	const modelValue = defineModel({default: true, type: Boolean})

	const methods = {
		handleClose() {
			emit('close')
		}
	}
</script>
