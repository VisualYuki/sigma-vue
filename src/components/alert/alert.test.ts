import {describe, expect, it} from 'vitest'
import {UiAlert} from './'
import {mount} from '@vue/test-utils'
import {colors} from '@/types/configuration'
import {alertVariants} from '@/components/alert/theme'
import {useNamespace} from '@/utils/use-namespace'

const ns = useNamespace('alert', true)

describe('', () => {
	const props = {title: 'custom title'}

	it.each([
		['with title', {props: {...props}}],
		['with description', {props: {...props, description: 'custom description'}}],
		[
			'overrideUi prop',
			{
				props: {
					...props,
					overrideUi: {
						root: 'override-root',
						title: 'override-title',
						wrapper: 'override-wrapper',
						close: 'override-close'
					}
				}
			}
		],
		...colors.map((color: string) => {
			return [`color: ${color}`, {props: {...props, color: color}}]
		}),
		...alertVariants.map((variant: string) => {
			return [`variant: ${variant}`, {props: {...props, variant: variant}}]
		}),
		['closable prop', {props: {closable: true}}]
		//@ts-ignore
	])('%s', (name: string, {props}) => {
		const wrapper = mount(UiAlert, {props})

		expect(wrapper.html()).toMatchSnapshot()
	})

	it('closable prop', async () => {
		const wrapper = mount(UiAlert, {props: {closable: true}})

		await wrapper.find(ns.e('close')).trigger('click')

		expect(wrapper.emitted()).toHaveProperty('close')
		expect(wrapper.emitted()).toHaveProperty('update:modelValue')
	})

	it('v-model', async () => {
		const wrapper = mount(UiAlert, {
			props: {
				modelValue: true,
				'onUpdate:modelValue': (e) => wrapper.setProps({modelValue: e}),
				closable: true
			}
		})

		expect(wrapper.props('modelValue')).toBe(true)

		await wrapper.find(ns.e('close')).trigger('click')
		expect(wrapper.props('modelValue')).toBe(false)

		await wrapper.setProps({
			modelValue: false
		})
		expect(wrapper.props('modelValue')).toBe(false)
	})

	// it('title prop & description prop', async () => {
	// 	const wrapper = mount(UiAlert, {
	// 		props: {
	// 			title: 'title1'
	// 		}
	// 	})

	// 	expect(wrapper.find('.' + ns.em('title', 'with-description')).exists()).toBe(false)
	// 	expect(wrapper.find('.' + ns.e('title')).exists()).toBe(true)
	// 	expect(wrapper.find('.' + ns.e('description')).exists()).toBe(false)

	// 	await wrapper.setProps({
	// 		description: 'description 1'
	// 	})

	// 	expect(wrapper.find('.' + ns.em('title', 'with-description')).exists()).toBe(true)
	// 	expect(wrapper.find('.' + ns.e('description')).exists()).toBe(true)
	// })

	// it('showIcon prop', async () => {
	// 	const wrapper = mount(UiAlert, {
	// 		props: {
	// 			center: true
	// 		}
	// 	})

	// 	wrapper.find('.' + ns.e('center'))
	// })
})
