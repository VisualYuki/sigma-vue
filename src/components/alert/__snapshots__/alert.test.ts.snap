// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`closable prop 1`] = `
"<transition-stub appear="false" persisted="false" css="true">
  <!-- TODO -->
  <div class="transition-collapse-target">
    <div class="relative overflow-hidden w-full rounded-(--ui-radius) p-4 items-center bg-(--ui-primary) text-(--ui-text-contrast) ui-alert__root ui-alert_primary">
      <div class="flex items-center">
        <!--v-if-->
        <div class="flex flex-col gap-1">
          <!--v-if-->
          <!--v-if-->
        </div>
        <uiicon icon="iconoir:cancel" class="hover:text-red-500 default-transition absolute top-3 right-3 cursor-pointer ui-alert__close" size="20"></uiicon>
      </div>
    </div>
  </div>
</transition-stub>"
`;

exports[`color: error 1`] = `
"<transition-stub appear="false" persisted="false" css="true">
  <!-- TODO -->
  <div class="transition-collapse-target">
    <div class="relative overflow-hidden w-full rounded-(--ui-radius) p-4 items-center bg-(--ui-error) text-(--ui-text-contrast) ui-alert__root ui-alert_error">
      <div class="flex items-center">
        <!--v-if-->
        <div class="flex flex-col gap-1">
          <div class="text-sm font-medium pe-6 ui-alert__title">custom title</div>
          <!--v-if-->
        </div>
        <!--v-if-->
      </div>
    </div>
  </div>
</transition-stub>"
`;

exports[`color: info 1`] = `
"<transition-stub appear="false" persisted="false" css="true">
  <!-- TODO -->
  <div class="transition-collapse-target">
    <div class="relative overflow-hidden w-full rounded-(--ui-radius) p-4 items-center bg-(--ui-info) text-(--ui-text-contrast) ui-alert__root ui-alert_info">
      <div class="flex items-center">
        <!--v-if-->
        <div class="flex flex-col gap-1">
          <div class="text-sm font-medium pe-6 ui-alert__title">custom title</div>
          <!--v-if-->
        </div>
        <!--v-if-->
      </div>
    </div>
  </div>
</transition-stub>"
`;

exports[`color: primary 1`] = `
"<transition-stub appear="false" persisted="false" css="true">
  <!-- TODO -->
  <div class="transition-collapse-target">
    <div class="relative overflow-hidden w-full rounded-(--ui-radius) p-4 items-center bg-(--ui-primary) text-(--ui-text-contrast) ui-alert__root ui-alert_primary">
      <div class="flex items-center">
        <!--v-if-->
        <div class="flex flex-col gap-1">
          <div class="text-sm font-medium pe-6 ui-alert__title">custom title</div>
          <!--v-if-->
        </div>
        <!--v-if-->
      </div>
    </div>
  </div>
</transition-stub>"
`;

exports[`color: secondary 1`] = `
"<transition-stub appear="false" persisted="false" css="true">
  <!-- TODO -->
  <div class="transition-collapse-target">
    <div class="relative overflow-hidden w-full rounded-(--ui-radius) p-4 items-center bg-(--ui-secondary) text-(--ui-text-contrast) ui-alert__root ui-alert_secondary">
      <div class="flex items-center">
        <!--v-if-->
        <div class="flex flex-col gap-1">
          <div class="text-sm font-medium pe-6 ui-alert__title">custom title</div>
          <!--v-if-->
        </div>
        <!--v-if-->
      </div>
    </div>
  </div>
</transition-stub>"
`;

exports[`color: success 1`] = `
"<transition-stub appear="false" persisted="false" css="true">
  <!-- TODO -->
  <div class="transition-collapse-target">
    <div class="relative overflow-hidden w-full rounded-(--ui-radius) p-4 items-center bg-(--ui-success) text-(--ui-text-contrast) ui-alert__root ui-alert_success">
      <div class="flex items-center">
        <!--v-if-->
        <div class="flex flex-col gap-1">
          <div class="text-sm font-medium pe-6 ui-alert__title">custom title</div>
          <!--v-if-->
        </div>
        <!--v-if-->
      </div>
    </div>
  </div>
</transition-stub>"
`;

exports[`color: warning 1`] = `
"<transition-stub appear="false" persisted="false" css="true">
  <!-- TODO -->
  <div class="transition-collapse-target">
    <div class="relative overflow-hidden w-full rounded-(--ui-radius) p-4 items-center bg-(--ui-warning) text-(--ui-text-contrast) ui-alert__root ui-alert_warning">
      <div class="flex items-center">
        <!--v-if-->
        <div class="flex flex-col gap-1">
          <div class="text-sm font-medium pe-6 ui-alert__title">custom title</div>
          <!--v-if-->
        </div>
        <!--v-if-->
      </div>
    </div>
  </div>
</transition-stub>"
`;

exports[`overrideUi prop 1`] = `
"<transition-stub appear="false" persisted="false" css="true">
  <!-- TODO -->
  <div class="transition-collapse-target">
    <div class="relative overflow-hidden w-full rounded-(--ui-radius) p-4 items-center bg-(--ui-primary) text-(--ui-text-contrast) override-root ui-alert__root ui-alert_primary">
      <div class="flex items-center override-wrapper">
        <!--v-if-->
        <div class="flex flex-col gap-1">
          <div class="text-sm font-medium pe-6 override-title ui-alert__title">custom title</div>
          <!--v-if-->
        </div>
        <!--v-if-->
      </div>
    </div>
  </div>
</transition-stub>"
`;

exports[`variant: outline 1`] = `
"<transition-stub appear="false" persisted="false" css="true">
  <!-- TODO -->
  <div class="transition-collapse-target">
    <div class="relative overflow-hidden w-full rounded-(--ui-radius) p-4 items-center text-(--ui-primary) ring ring-inset ring-(--ui-primary)/25 ui-alert__root ui-alert_primary">
      <div class="flex items-center">
        <!--v-if-->
        <div class="flex flex-col gap-1">
          <div class="text-sm font-medium pe-6 ui-alert__title">custom title</div>
          <!--v-if-->
        </div>
        <!--v-if-->
      </div>
    </div>
  </div>
</transition-stub>"
`;

exports[`variant: soft 1`] = `
"<transition-stub appear="false" persisted="false" css="true">
  <!-- TODO -->
  <div class="transition-collapse-target">
    <div class="relative overflow-hidden w-full rounded-(--ui-radius) p-4 items-center bg-(--ui-primary)/10 text-(--ui-primary) ui-alert__root ui-alert_primary">
      <div class="flex items-center">
        <!--v-if-->
        <div class="flex flex-col gap-1">
          <div class="text-sm font-medium pe-6 ui-alert__title">custom title</div>
          <!--v-if-->
        </div>
        <!--v-if-->
      </div>
    </div>
  </div>
</transition-stub>"
`;

exports[`variant: solid 1`] = `
"<transition-stub appear="false" persisted="false" css="true">
  <!-- TODO -->
  <div class="transition-collapse-target">
    <div class="relative overflow-hidden w-full rounded-(--ui-radius) p-4 items-center bg-(--ui-primary) text-(--ui-text-contrast) ui-alert__root ui-alert_primary">
      <div class="flex items-center">
        <!--v-if-->
        <div class="flex flex-col gap-1">
          <div class="text-sm font-medium pe-6 ui-alert__title">custom title</div>
          <!--v-if-->
        </div>
        <!--v-if-->
      </div>
    </div>
  </div>
</transition-stub>"
`;

exports[`variant: subtle 1`] = `
"<transition-stub appear="false" persisted="false" css="true">
  <!-- TODO -->
  <div class="transition-collapse-target">
    <div class="relative overflow-hidden w-full rounded-(--ui-radius) p-4 items-center bg-(--ui-primary)/10 text-(--ui-primary) ring ring-inset ring-(--ui-primary)/25 ui-alert__root ui-alert_primary">
      <div class="flex items-center">
        <!--v-if-->
        <div class="flex flex-col gap-1">
          <div class="text-sm font-medium pe-6 ui-alert__title">custom title</div>
          <!--v-if-->
        </div>
        <!--v-if-->
      </div>
    </div>
  </div>
</transition-stub>"
`;

exports[`with description 1`] = `
"<transition-stub appear="false" persisted="false" css="true">
  <!-- TODO -->
  <div class="transition-collapse-target">
    <div class="relative overflow-hidden w-full rounded-(--ui-radius) p-4 items-center bg-(--ui-primary) text-(--ui-text-contrast) ui-alert__root ui-alert_primary">
      <div class="flex items-center">
        <!--v-if-->
        <div class="flex flex-col gap-1">
          <div class="text-sm font-medium pe-6 ui-alert__title">custom title</div>
          <div class="text-sm opacity-90 ui-alert__description">custom description</div>
        </div>
        <!--v-if-->
      </div>
    </div>
  </div>
</transition-stub>"
`;

exports[`with title 1`] = `
"<transition-stub appear="false" persisted="false" css="true">
  <!-- TODO -->
  <div class="transition-collapse-target">
    <div class="relative overflow-hidden w-full rounded-(--ui-radius) p-4 items-center bg-(--ui-primary) text-(--ui-text-contrast) ui-alert__root ui-alert_primary">
      <div class="flex items-center">
        <!--v-if-->
        <div class="flex flex-col gap-1">
          <div class="text-sm font-medium pe-6 ui-alert__title">custom title</div>
          <!--v-if-->
        </div>
        <!--v-if-->
      </div>
    </div>
  </div>
</transition-stub>"
`;
